toolchain/check/type.h

```diff
@@ -108,19 +108,40 @@ auto GetUnboundElementType(Context& context, SemIR::TypeInstId class_type_id,
                            SemIR::TypeInstId element_type_id) -> SemIR::TypeId;
 
 // Convert a facet value or type value instruction to a canonical facet or type
-// value instruction.
+// value instruction. This finds a canonical value across instructions of type
+// `TypeType` and of type `FacetType`.
 //
 // Type values are already canonical and are returned unchanged, except for
 // `FacetAccessType` which is unwrapped to find the facet value it refers to.
 //
-// For facet values, unwraps `FacetValue` instructions to get to an underlying
-// canonical type instruction.
-auto GetCanonicalizedFacetOrTypeValue(Context& context, SemIR::InstId inst_id)
+// If `preserve_facet_value` is false, the `FacetValue` instruction is unwrapped
+// to get to an underlying canonical type instruction. Otherwise, the output is
+// either the original instruction or a facet value instruction.
```

Comment:
Maybe I'm misunderstanding something, but it looks to me like this comment doesn't match the implementation. Specifically, I think if we're given a `SymbolicBindingType`, we'll always produce a facet value regardless of `preserve_facet_value`, and won't produce a canoncial type instruction (which would be the `SymbolicBindingType`).